client/client.go

```diff
@@ -253,30 +245,39 @@ func (c httpMetricClient) retryWithBackoff(
 	for i := 0; i < c.maxRetries; i++ {
 
 		var uploadURL, hash string
-		uploadURL, hash, err = c.GetUploadURL(metricFile, metricSampleURL, agentVersion, UID)
+		uploadURL, hash, err = c.GetUploadURL(metricFile, metricSampleURL, agentVersion, UID, i)
 		if err != nil {
 			log.Debugf("Client proxy or deployment YAML may be misconfigured.  Please check your client settings.")
 			log.Errorf("error encountered while retrieving upload location: %v", err)
 			continue
 		}
 
+		var awsRequestID, statusMessage string
+		var responseDump []byte
+		var dumpErr error
 		resp, err = c.buildAndDoRequest(metricFile, uploadURL, agentVersion, UID, hash)
-
-		if c.verbose {
-			reponseDump, err := httputil.DumpResponse(resp, true)
-			if err != nil {
-				log.Errorln(err)
+		if resp != nil {
+			awsRequestID = resp.Header.Get("X-Amz-Request-Id")
+			statusMessage = resp.Status
+			responseDump, dumpErr = httputil.DumpResponse(resp, true)
+			if dumpErr != nil {
+				log.Errorln(dumpErr)
```

Comment:
I'm not sold on keeping the `continue` in here, if the request succeeds then I don't think we care if the response failed